app/routes.js

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import { Route, IndexRoute } from 'react-router';
+
+import AppComponent from './components/app';
+import Patients from './pages/patients';
+import Login from './pages/login';
+import Signup from './pages/signup';
+import Profile from './pages/profile';
+import Patient from './pages/patient';
+import PatientNew from './pages/patientnew';
+import PatientData from './pages/patientdata';
+import RequestPasswordReset from './pages/passwordreset/request';
+import ConfirmPasswordReset from './pages/passwordreset/confirm';
+import EmailVerification from './pages/emailverification';
+
+import personUtils from './core/personutils';
+
+/**
+ * This function redirects any requests that land on pages that should only be
+ * visible when logged in if the user is logged out
+ * 
+ * @param  {Object} nextState
+ * @param  {Function} replaceState
+ *
+ * @return {boolean|null} returns true if hash mapping happened
+ */
+export const requireAuth = (api) => (nextState, replaceState) => {
+  if (!api.user.isAuthenticated()) {
+    replaceState(null, '/login');
+  }
+};
+
+/**
+ * This function redirects any requests that land on pages that should only be
+ * visible when no data storage is set up if the user has data storage set up
+ *
+ * @param  {Object} nextState
+ * @param  {Function} replaceState
+ *
+ * @return {boolean|null} returns true if hash mapping happened
+ */
+export const requireAuthAndNoPatient = (api, cb) => (nextState, replaceState) => {
+  if (!api.user.isAuthenticated()) {
+    replaceState(null, '/login');
+    cb();
+  }
+  else {
+    api.user.get(function(err, user) {
+      if (personUtils.isPatient(user)) {
+        replaceState(null, '/patients');
+        cb();
+      }
+      cb();
+    });
+  }
+};
+
+/**
+ * This function redirects any requests that land on pages that should only be
+ * visible when logged out if the user is logged in
+ * 
+ * @param  {Object} nextState
+ * @param  {Function} replaceState
+ *
+ * @return {boolean|null} returns true if hash mapping happened
+ */
+export const requireNoAuth = (api) => (nextState, replaceState) => {
+  if (api.user.isAuthenticated()) {
+    replaceState(null, '/patients');
+  }
+};
+
+/**
+ * This function redirects any requests that land on pages that should only be
+ * visible when the user hasn't yet verified their sign-up e-mail
+ * if the user already has completed the e-mail verification
+ *
+ * @param  {Object} nextState
+ * @param  {Function} replaceState
+ *
+ * @return {boolean|null} returns true if hash mapping happened
+ */
+export const requireNotVerified = (api, cb) => (nextState, replaceState) => {
```

Comment:
@GordyD and here's the second one, because we weren't getting the proper flow through /email-verification

(This also required a backend change to return the `emailVerified` boolean with the user metadata.)